text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,415 @@
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc:component>`. The tag name used for the dynamic component is the value returned from `getComponentRegisteredName`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+Dynamic components have certain properties that will not work with `this.template.querySelector`:
+1. The `<lwc:component>` tag will not be rendered to the DOM which means it is not available to be selected using `this.template.querySelector`. 
+2. The LWC engine does not provide a way to retrieve the dynamic component's custom element name for use as a selector with `this.template.querySelector` (we recommend using `lwc:ref` instead).
+
+To select the dynamic component, the actual custom element must be selected once it has been rendered to the DOM by using the [`lwc:ref`](https://rfcs.lwc.dev/rfcs/lwc/0000-refs) directive or through another attribute assigned to the component.
+
+Here are some ways component authors can detect when a dynamic component has mounted:
+- Use `connectedCallback` on the dynamic component to signal when it has mounted.
+- A dynamic component constructor is guaranteed to be mounted in the next rendering cycle once it has been set.  When it is set, the parent component can wait until the `renderedCallback` lifecycle method is invoked to detect when the dynamic component is mounted.
+- Use a [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to detect when the dynamic component has mounted.
+
+```html
+<template>
+    <lwc:component lwc:is={lazyConstructor} lwc:ref="foo"></lwc:component>
+</template>
+```
+
+```javascript
+import { LightningElement } from 'lwc';
+
+export default class extends LightningElement {
+    lazyConstructor;
+
+    connectedCallback() {
+        import('lightning/concreteComponent')
+            .then(({ default: ctor }) => this.lazyConstructor = ctor)
+            .catch(err => console.log('Error importing component'));
+    }
+
+    renderedCallback() {
+        // this.refs.foo will be available on the next rendering cycle after the constructor has been set.
+        if (this.refs.foo) {
+            // this.refs.foo will contain a reference to the DOM node
+            console.log(this.refs.foo);
+        }
+    }
+}
+```
+
+#### Assigning attributes and event listeners
+
+Attributes and event listeners can be passed to the `<lwc:component>` tag declaratively or can be assigned directly to the custom element imperatively.
```

Comment:
I think the whole section is kinda not needed, just mentioned that `<lwc:component>` does accept all , or almost all APIs (I believe it supports all). And it obey the roles of those apis, including attributes, events, etc.